### PR TITLE
Microphysics package with saturation adjustment

### PR DESCRIPTION
**Pull request by OngChia**
Add saturation adjustment component to icon4py. All future physics will be placed under `subgrid_scale_phyiscs` folder.
1. Saturation adjustment component has a `run` function that returns the tendencies, conforming with the decision made in this PR https://github.com/C2SM/icon4py/pull/506. However, a follow-up action needs to be done in a subsequent PR to make this component fully follow the protocol. 
2. A datatest is added to test the port of saturation adjustment using the serialized data generated from the Weisman-Klemp experiment run on the Torus grid.
3. Add savepoints for datatest of both graupel microphysics and saturation adjustment since the same experiment is used for serializing the data in graupel and saturation adjustment calls.

Some additional fixes are:
1. add tracer state.
2. the computations in diagnose_pressure, diagnose_temperature, and diagnose_surface_pressure are changed accordingly as we now have moisture-related tracers.
3. Changes import statement in the above stencils and other stencils used in icon4py driver initialization.

### CODE REVIEW

File: model/atmosphere/subgrid_scale_physics/microphysics/src/icon4py/model/atmosphere/subgrid_scale_physics/microphysics/saturation_adjustment.py [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+from icon4py.model.common.states import (
+    diagnostic_state as diagnostics,
+    prognostic_state as prognostics,
+    tracer_state as tracers,
+)
+from icon4py.model.common.utils import gt4py_field_allocation as field_alloc
+
+
+# TODO (Chia Rui): Refactor this class when there is consensus in gt4py team about the best way to express compile-time constants
+class ConvectTables(FrozenNamespace):
```
[the remark this answers is not in the document]
Comment (halungge) [reply model/atmosphere/subgrid_scale_physics/microphysics/src/icon4py/model/atmosphere/subgrid_scale_physics/microphysics/saturation_adjustment.py]: I think Nikki was working on that: https://github.com/GridTools/gt4py/pull/1615 would that already fix the issue?

**Comment by halungge** [comment]
> Perhaps I will leave the saturaton_adjustment component as it is now in this PR. And I found that the input and output of graupel microphysics have just slightly more, but similar, arguments than saturation adjustment. The progress there is good now. I will open another PR after graupel merge to make these two components follow the interface protocol and more seriously think about integration into the driver. How is this plan?

That is fine for me.

File: model/common/src/icon4py/model/common/interpolation/stencils/edge_2_cell_vector_rbf_interpolation.py
[the remark this answers is not in the document]
Comment (halungge) [reply model/common/src/icon4py/model/common/interpolation/stencils/edge_2_cell_vector_rbf_interpolation.py:1]: Ah so the interpolation coefficent that is used is generated by some RBF functions. But then would you agree that as the coefficients are arguments to the stencils (so any suitable interpolation coefficient, produced by whatever method) and the connectivity is just a connectivity, the stencil function itself independent of RBF?

We should maybe rename (not necessarily in this PR).

File: model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/init_exner_pr.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
```
[the remark this answers is not in the document]
Comment (halungge) [reply model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/init_exner_pr.py]: Liked the new name. Thank you. I always thought `exner_pr` stand for `exner_pressure` and was wondering why and what the difference was to the `exner` which is also exner pressure... Never guessed that `_pr` stands for "perturbation"... so much for the abbreviations....  :see_no_evil:

File: model/common/src/icon4py/model/common/test_utils/datatest_utils.py [hunk 3]
```diff
@@ -20,19 +20,23 @@
...
 GAUSS3D_EXPERIMENT = "gauss3d_torus"
+WEISMAN_KLEMP_EXPERIMENT = "weisman_klemp_torus"
 
 MC_CH_R04B09_DSL_GRID_URI = "https://polybox.ethz.ch/index.php/s/hD232znfEPBh4Oh/download"
 R02B04_GLOBAL_GRID_URI = "https://polybox.ethz.ch/index.php/s/AKAO6ImQdIatnkB/download"
+TORUS_100X116_1000M_GRID_URI = "https://polybox.ethz.ch/index.php/s/yqvotFss9i1OKzs/download"
 GRID_URIS = {
     REGIONAL_EXPERIMENT: MC_CH_R04B09_DSL_GRID_URI,
     R02B04_GLOBAL: R02B04_GLOBAL_GRID_URI,
+    WEISMAN_KLEMP_EXPERIMENT: TORUS_100X116_1000M_GRID_URI,
```
[the remark this answers is not in the document]
Comment (halungge) [reply model/common/src/icon4py/model/common/test_utils/datatest_utils.py:32]: ok

**Review by halungge — APPROVED** [review]
I left some remarks, (more discussion points...) 
For me is fine to leave the component interface to a separate PR.

File: model/common/src/icon4py/model/common/interpolation/stencils/edge_2_cell_vector_rbf_interpolation.py
[the remark this answers is not in the document]
Comment (OngChia) [reply model/common/src/icon4py/model/common/interpolation/stencils/edge_2_cell_vector_rbf_interpolation.py:1]: Highly agreed. I will rename it in my subsequent PR.

**Comment by OngChia** [comment]
cscs-ci run default

**Comment by OngChia** [comment]
launch jenkins spack

**Pull request merged**